lib/Map/DataVariable.js

```diff
@@ -65,29 +80,16 @@ DataVariable.prototype.update = function () {
 
 // Based on variable name, try to determine best default for varType
 DataVariable.prototype._guessVariableType = function () {
-    //functions to try to figure out position and time variables.
-    function matchColumn(name, hints) {
-        name = name.toLowerCase();
-        for (var h in hints) {
-            if (hints.hasOwnProperty(h)) {
-                var hint = hints[h].toLowerCase();
-                if (name.indexOf(hint) === 0 || name.indexOf(' ' + hint) !== -1 || name.indexOf('_' + hint) !== -1) {
-                    return true;
-                }
-            }
-        }
-        return false;
-    }
-
     var hintSet = [
-        { hints: ['lon'], type: VarType.LON },
-        { hints: ['lat'], type: VarType.LAT },
-        { hints: ['depth', 'height', 'elevation'], type: VarType.ALT },
-        { hints: ['time', 'date'], type: VarType.TIME }];
-
-    for (var vt in hintSet) {
-        if (matchColumn(this.name, hintSet[vt].hints)) {
-            this.varType = hintSet[vt].type;
+        { hint: /^(lon|longitude|lng)$/i, type: VarType.LON },
+        { hint: /^(lat|latitude)$/i, type: VarType.LAT },
+        { hint: /^(.*[_ ])?(depth|height|elevation)$/i, type: VarType.ALT },
+        { hint: /^(.*[_ ])?(time|date)$/i, type: VarType.TIME },
+        { hint: /^postcode|poa|(.*_code)$/i, type: VarType.ENUM }]; //## This doesn't actually prevent misparsing.
```

Comment:
I was trying to figure out why numeric codes were being coerced into integers. Figured it out. Fixed it. Comment gone now.